src/registry/mod.rs

```diff
@@ -108,4 +109,41 @@ impl RegistryUri {
 
         Ok(())
     }
+
+    /// Returns the type of registry this URI refers to
+    pub fn kind(&self) -> RegistryType {
+        RegistryType::Artifactory
+    }
+}
+
+#[derive(Default)]
+pub struct RegistryProvider {
+    credentials: Credentials,
+    registries: HashMap<RegistryUri, Arc<dyn Registry + Send + Sync>>,
+}
+
+impl RegistryProvider {
+    pub fn new(credentials: Credentials) -> Self {
+        Self {
+            credentials,
+            registries: Default::default(),
+        }
+    }
+
+    pub fn get_or_create(
+        &mut self,
+        registry_uri: &RegistryUri,
```

Comment:
```suggestion
        uri: &RegistryUri,
```

(to keep this the same as everywhere else)